
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://genestorian.github.io/OpenCloning_LinkML/">
      
      <link rel="icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-8.5.10">
    
    
      
        <title>OpenCloning_LinkML</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.975780f9.min.css">
      
        
        <link rel="stylesheet" href="assets/stylesheets/palette.2505c338.min.css">
        
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="None" data-md-color-accent="None">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#opencloning_linkml" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="." title="OpenCloning_LinkML" class="md-header__button md-logo" aria-label="OpenCloning_LinkML" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            OpenCloning_LinkML
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Index
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/genestorian/OpenCloning_LinkML" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="." title="OpenCloning_LinkML" class="md-nav__button md-logo" aria-label="OpenCloning_LinkML" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    OpenCloning_LinkML
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/genestorian/OpenCloning_LinkML" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Index
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="." class="md-nav__link md-nav__link--active">
        Index
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    Classes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    Slots
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    Enumerations
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    Types
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    Subsets
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="about/" class="md-nav__link">
        About
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    Classes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    Slots
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    Enumerations
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    Types
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    Subsets
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  <a href="https://github.com/genestorian/OpenCloning_LinkML/edit/master/docs/index.md" title="Edit this page" class="md-content__button md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>


<h1 id="opencloning_linkml">OpenCloning_LinkML</h1>
<p>A LinkML data model for OpenCloning</p>
<p>URI: https://w3id.org/genestorian/OpenCloning_LinkML</p>
<p>Name: OpenCloning_LinkML</p>
<h2 id="classes">Classes</h2>
<table>
<thead>
<tr>
<th>Class</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="AnnotationReport/">AnnotationReport</a></td>
<td>Represents a report of an annotation step</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="PlannotateAnnotationReport/">PlannotateAnnotationReport</a></td>
<td>Represents a report of an annotation step using Plannotate</td>
</tr>
<tr>
<td><a href="AssemblyFragment/">AssemblyFragment</a></td>
<td>Represents a fragment in an assembly</td>
</tr>
<tr>
<td><a href="AssociatedFile/">AssociatedFile</a></td>
<td>Represents a file associated with a sequence</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="SequencingFile/">SequencingFile</a></td>
<td>Represents a sequencing file and its alignment to a sequence</td>
</tr>
<tr>
<td><a href="CloningStrategy/">CloningStrategy</a></td>
<td>Represents a cloning strategy</td>
</tr>
<tr>
<td><a href="CollectionOption/">CollectionOption</a></td>
<td>Represents an option in a collection</td>
</tr>
<tr>
<td><a href="CollectionOptionInfo/">CollectionOptionInfo</a></td>
<td>Additional information about a collection option</td>
</tr>
<tr>
<td><a href="NamedThing/">NamedThing</a></td>
<td>None</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="Sequence/">Sequence</a></td>
<td>Represents a sequence</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="Primer/">Primer</a></td>
<td>An oligonucleotide or primer</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="TemplateSequence/">TemplateSequence</a></td>
<td>Represents a sequence that is part of a template, where the actual sequence content will be determined by the user's actions</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="TextFileSequence/">TextFileSequence</a></td>
<td>A sequence (may have features) defined by the content of a text file</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="Source/">Source</a></td>
<td>Represents the source of a sequence</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="AnnotationSource/">AnnotationSource</a></td>
<td>Represents a computational step in which sequence features are annotated in a sequence</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="AssemblySource/">AssemblySource</a></td>
<td>Represents the source of a sequence that is an assembly of other sequences</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="GatewaySource/">GatewaySource</a></td>
<td>Represents the source of a sequence that is generated by Gateway cloning</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="GibsonAssemblySource/">GibsonAssemblySource</a></td>
<td>Represents the source of a sequence that is generated by Gibson assembly</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="HomologousRecombinationSource/">HomologousRecombinationSource</a></td>
<td>Represents the source of a sequence that is generated by homologous recombination</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="CRISPRSource/">CRISPRSource</a></td>
<td>Represents the source of a sequence that is generated by CRISPR</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="InFusionSource/">InFusionSource</a></td>
<td>Represents the source of a sequence that is generated by In-Fusion cloning by Takara Bio</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="LigationSource/">LigationSource</a></td>
<td>Represents the source of a sequence that is generated by ligation with sticky or blunt ends.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="OverlapExtensionPCRLigationSource/">OverlapExtensionPCRLigationSource</a></td>
<td>Represents the source of a sequence that is generated by ligation of PCR products as part of overlap extension PCR. Algorithmically equivalent to Gibson assembly.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="PCRSource/">PCRSource</a></td>
<td>Represents the source of a sequence that is generated by PCR</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="RestrictionAndLigationSource/">RestrictionAndLigationSource</a></td>
<td>Represents the source of a sequence that is generated by restriction and ligation</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="CollectionSource/">CollectionSource</a></td>
<td>Represents a collection of possible sources in a template</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="GenomeCoordinatesSource/">GenomeCoordinatesSource</a></td>
<td>Represents the source of a sequence that is identified by genome coordinates, requested from NCBI</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ManuallyTypedSource/">ManuallyTypedSource</a></td>
<td>Represents the source of a sequence that is manually typed by the user</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="OligoHybridizationSource/">OligoHybridizationSource</a></td>
<td>Represents the source of a sequence that is generated by oligo hybridization</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="PolymeraseExtensionSource/">PolymeraseExtensionSource</a></td>
<td>Represents the source of a sequence that is generated by polymerase extension</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="RepositoryIdSource/">RepositoryIdSource</a></td>
<td>Represents the source of a sequence that is identified by a repository id</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="AddGeneIdSource/">AddGeneIdSource</a></td>
<td>Represents the source of a sequence that is identified by an AddGene id</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="BenchlingUrlSource/">BenchlingUrlSource</a></td>
<td>Represents the source of a sequence that is identified by a Benchling URL</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="EuroscarfSource/">EuroscarfSource</a></td>
<td>Represents the source of a sequence from the Euroscarf plasmid library</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="IGEMSource/">IGEMSource</a></td>
<td>Represents the source of a sequence from an iGEM collection</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="SEVASource/">SEVASource</a></td>
<td>Represents the source of a sequence that is identified by a SEVA id</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="SnapGenePlasmidSource/">SnapGenePlasmidSource</a></td>
<td>Represents the source of a sequence from the SnapGene plasmid library identified by a SnapGene subpath of https://www.snapgene.com/plasmids/</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="WekWikGeneIdSource/">WekWikGeneIdSource</a></td>
<td>Represents the source of a sequence that is identified by a WekWikGene id</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="ReverseComplementSource/">ReverseComplementSource</a></td>
<td>Represents the in-silico transformation of a sequence into its reverse complement</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="SequenceCutSource/">SequenceCutSource</a></td>
<td>Represents the source of a sequence that is a subfragment of another sequence, generated by sequence cutting.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="RestrictionEnzymeDigestionSource/">RestrictionEnzymeDigestionSource</a></td>
<td>Represents the source of a sequence that is a subfragment of another sequence, generated by sequence cutting using restriction enzymes.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="UploadedFileSource/">UploadedFileSource</a></td>
<td>Represents the source of a sequence that is uploaded as a file</td>
</tr>
<tr>
<td><a href="SequenceCut/">SequenceCut</a></td>
<td>Represents a cut in a DNA sequence</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="RestrictionSequenceCut/">RestrictionSequenceCut</a></td>
<td>Represents a cut in a DNA sequence that is made by a restriction enzyme</td>
</tr>
<tr>
<td><a href="SimpleSequenceLocation/">SimpleSequenceLocation</a></td>
<td>Represents a location within a sequence, for now support for ranges only</td>
</tr>
</tbody>
</table>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Slot</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="add_primer_features/">add_primer_features</a></td>
<td>Whether to add primer features to the PCR product</td>
</tr>
<tr>
<td><a href="addgene_sequence_type/">addgene_sequence_type</a></td>
<td></td>
</tr>
<tr>
<td><a href="alignment/">alignment</a></td>
<td>The alignment of the sequencing read to the sequence</td>
</tr>
<tr>
<td><a href="annotation_report/">annotation_report</a></td>
<td></td>
</tr>
<tr>
<td><a href="annotation_tool/">annotation_tool</a></td>
<td></td>
</tr>
<tr>
<td><a href="annotation_tool_version/">annotation_tool_version</a></td>
<td>The version of the annotation tool</td>
</tr>
<tr>
<td><a href="assembly/">assembly</a></td>
<td>A list of the fragments that are assembled, in order</td>
</tr>
<tr>
<td><a href="assembly_accession/">assembly_accession</a></td>
<td>The accession of the assembly</td>
</tr>
<tr>
<td><a href="category_id/">category_id</a></td>
<td>The identifier of the category of the part in the template</td>
</tr>
<tr>
<td><a href="circular/">circular</a></td>
<td>Whether the sequence is circular or linear</td>
</tr>
<tr>
<td><a href="circularize/">circularize</a></td>
<td>Whether the sequence should be circularized (FASTA only)</td>
</tr>
<tr>
<td><a href="cut_watson/">cut_watson</a></td>
<td>The position of the cut in the watson strand</td>
</tr>
<tr>
<td><a href="database/">database</a></td>
<td></td>
</tr>
<tr>
<td><a href="description/">description</a></td>
<td>A description of the category</td>
</tr>
<tr>
<td><a href="Description/">Description</a></td>
<td></td>
</tr>
<tr>
<td><a href="end/">end</a></td>
<td>The ending coordinate (1-based) of the sequence in the sequence accession</td>
</tr>
<tr>
<td><a href="end_location/">end_location</a></td>
<td></td>
</tr>
<tr>
<td><a href="Feature/">Feature</a></td>
<td></td>
</tr>
<tr>
<td><a href="file_content/">file_content</a></td>
<td></td>
</tr>
<tr>
<td><a href="file_name/">file_name</a></td>
<td>The name of the file</td>
</tr>
<tr>
<td><a href="file_type/">file_type</a></td>
<td>The type of file</td>
</tr>
<tr>
<td><a href="files/">files</a></td>
<td>Files associated with this cloning strategy</td>
</tr>
<tr>
<td><a href="forward_oligo/">forward_oligo</a></td>
<td>The forward oligo used in the hybridization</td>
</tr>
<tr>
<td><a href="fragment/">fragment</a></td>
<td></td>
</tr>
<tr>
<td><a href="full_length_of_feature_in_db/">full_length_of_feature_in_db</a></td>
<td></td>
</tr>
<tr>
<td><a href="gene_id/">gene_id</a></td>
<td>The gene id of the sequence</td>
</tr>
<tr>
<td><a href="greedy/">greedy</a></td>
<td>Whether to use a greedy consensus sequence for att sites (see https://github</td>
</tr>
<tr>
<td><a href="guides/">guides</a></td>
<td>The guide RNAs used in the CRISPR</td>
</tr>
<tr>
<td><a href="id/">id</a></td>
<td>A unique identifier for a thing</td>
</tr>
<tr>
<td><a href="image/">image</a></td>
<td>URL and size of the image representing this category</td>
</tr>
<tr>
<td><a href="index_in_file/">index_in_file</a></td>
<td>The index of the sequence in the file</td>
</tr>
<tr>
<td><a href="info/">info</a></td>
<td>Additional information about this option</td>
</tr>
<tr>
<td><a href="input/">input</a></td>
<td>The sequences that are an input to this source</td>
</tr>
<tr>
<td><a href="left_edge/">left_edge</a></td>
<td></td>
</tr>
<tr>
<td><a href="left_location/">left_location</a></td>
<td></td>
</tr>
<tr>
<td><a href="length_of_found_feature/">length_of_found_feature</a></td>
<td></td>
</tr>
<tr>
<td><a href="locus_tag/">locus_tag</a></td>
<td>The locus tag of the sequence</td>
</tr>
<tr>
<td><a href="name/">name</a></td>
<td>A human-readable name for a thing</td>
</tr>
<tr>
<td><a href="options/">options</a></td>
<td>The options available in this category</td>
</tr>
<tr>
<td><a href="output/">output</a></td>
<td>Identifier of the sequence that is the output of this source</td>
</tr>
<tr>
<td><a href="output_name/">output_name</a></td>
<td>Used to specify the name of the output sequence</td>
</tr>
<tr>
<td><a href="overhang/">overhang</a></td>
<td>The length of the overhang that is left after the cut</td>
</tr>
<tr>
<td><a href="overhang_crick_3prime/">overhang_crick_3prime</a></td>
<td>Taken from pydna's <code>dseq::ovhg</code>An integer describing the length of the crick ...</td>
</tr>
<tr>
<td><a href="overhang_watson_3prime/">overhang_watson_3prime</a></td>
<td>The equivalent of <code>overhang_crick_3prime</code> but for the watson strand</td>
</tr>
<tr>
<td><a href="percent_identity/">percent_identity</a></td>
<td></td>
</tr>
<tr>
<td><a href="percent_match_length/">percent_match_length</a></td>
<td></td>
</tr>
<tr>
<td><a href="primer_design/">primer_design</a></td>
<td>Can be used to indicate the intended primer design for this sequence in the t...</td>
</tr>
<tr>
<td><a href="primers/">primers</a></td>
<td>The primers that are used in the cloning strategy</td>
</tr>
<tr>
<td><a href="reaction_type/">reaction_type</a></td>
<td></td>
</tr>
<tr>
<td><a href="repository_id/">repository_id</a></td>
<td>The id of the sequence in the repository</td>
</tr>
<tr>
<td><a href="repository_name/">repository_name</a></td>
<td></td>
</tr>
<tr>
<td><a href="resistance/">resistance</a></td>
<td>The antibiotic resistance of the plasmid</td>
</tr>
<tr>
<td><a href="restriction_enzyme/">restriction_enzyme</a></td>
<td></td>
</tr>
<tr>
<td><a href="restriction_enzymes/">restriction_enzymes</a></td>
<td></td>
</tr>
<tr>
<td><a href="reverse_complemented/">reverse_complemented</a></td>
<td>Whether the sequence is reverse complemented in the assembly</td>
</tr>
<tr>
<td><a href="reverse_oligo/">reverse_oligo</a></td>
<td>The reverse oligo used in the hybridization</td>
</tr>
<tr>
<td><a href="right_edge/">right_edge</a></td>
<td></td>
</tr>
<tr>
<td><a href="right_location/">right_location</a></td>
<td></td>
</tr>
<tr>
<td><a href="sequence/">sequence</a></td>
<td></td>
</tr>
<tr>
<td><a href="sequence_accession/">sequence_accession</a></td>
<td>The accession of the sequence</td>
</tr>
<tr>
<td><a href="sequence_file_format/">sequence_file_format</a></td>
<td>The format of a sequence file</td>
</tr>
<tr>
<td><a href="sequence_file_url/">sequence_file_url</a></td>
<td>The URL of a sequence file</td>
</tr>
<tr>
<td><a href="sequence_id/">sequence_id</a></td>
<td>The sequence this file is associated with</td>
</tr>
<tr>
<td><a href="sequences/">sequences</a></td>
<td>The sequences that are used in the cloning strategy</td>
</tr>
<tr>
<td><a href="source/">source</a></td>
<td>The source of the sequence for this option</td>
</tr>
<tr>
<td><a href="sources/">sources</a></td>
<td>The sources of the sequences that are used in the cloning strategy</td>
</tr>
<tr>
<td><a href="sseqid/">sseqid</a></td>
<td></td>
</tr>
<tr>
<td><a href="start/">start</a></td>
<td>The starting coordinate (1-based) of the sequence in the sequence accession</td>
</tr>
<tr>
<td><a href="start_location/">start_location</a></td>
<td></td>
</tr>
<tr>
<td><a href="strand/">strand</a></td>
<td>The strand of the sequence in the sequence accession, should be 1 or -1</td>
</tr>
<tr>
<td><a href="title/">title</a></td>
<td>The title of the category</td>
</tr>
<tr>
<td><a href="type/">type</a></td>
<td>Designates the class</td>
</tr>
<tr>
<td><a href="Type/">Type</a></td>
<td></td>
</tr>
<tr>
<td><a href="user_input/">user_input</a></td>
<td></td>
</tr>
<tr>
<td><a href="well/">well</a></td>
<td>The well position in the kit plate</td>
</tr>
</tbody>
</table>
<h2 id="enumerations">Enumerations</h2>
<table>
<thead>
<tr>
<th>Enumeration</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="AddGeneSequenceType/">AddGeneSequenceType</a></td>
<td></td>
</tr>
<tr>
<td><a href="AnnotationTool/">AnnotationTool</a></td>
<td></td>
</tr>
<tr>
<td><a href="AssociatedFileType/">AssociatedFileType</a></td>
<td></td>
</tr>
<tr>
<td><a href="Collection/">Collection</a></td>
<td></td>
</tr>
<tr>
<td><a href="CollectionOptionType/">CollectionOptionType</a></td>
<td></td>
</tr>
<tr>
<td><a href="GatewayReactionType/">GatewayReactionType</a></td>
<td></td>
</tr>
<tr>
<td><a href="RepositoryName/">RepositoryName</a></td>
<td></td>
</tr>
<tr>
<td><a href="SequenceFileFormat/">SequenceFileFormat</a></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="types">Types</h2>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="Boolean/">Boolean</a></td>
<td>A binary (true or false) value</td>
</tr>
<tr>
<td><a href="Curie/">Curie</a></td>
<td>a compact URI</td>
</tr>
<tr>
<td><a href="Date/">Date</a></td>
<td>a date (year, month and day) in an idealized calendar</td>
</tr>
<tr>
<td><a href="DateOrDatetime/">DateOrDatetime</a></td>
<td>Either a date or a datetime</td>
</tr>
<tr>
<td><a href="Datetime/">Datetime</a></td>
<td>The combination of a date and time</td>
</tr>
<tr>
<td><a href="Decimal/">Decimal</a></td>
<td>A real number with arbitrary precision that conforms to the xsd:decimal speci...</td>
</tr>
<tr>
<td><a href="Double/">Double</a></td>
<td>A real number that conforms to the xsd:double specification</td>
</tr>
<tr>
<td><a href="Float/">Float</a></td>
<td>A real number that conforms to the xsd:float specification</td>
</tr>
<tr>
<td><a href="Integer/">Integer</a></td>
<td>An integer</td>
</tr>
<tr>
<td><a href="Jsonpath/">Jsonpath</a></td>
<td>A string encoding a JSON Path</td>
</tr>
<tr>
<td><a href="Jsonpointer/">Jsonpointer</a></td>
<td>A string encoding a JSON Pointer</td>
</tr>
<tr>
<td><a href="Ncname/">Ncname</a></td>
<td>Prefix part of CURIE</td>
</tr>
<tr>
<td><a href="Nodeidentifier/">Nodeidentifier</a></td>
<td>A URI, CURIE or BNODE that represents a node in a model</td>
</tr>
<tr>
<td><a href="Objectidentifier/">Objectidentifier</a></td>
<td>A URI or CURIE that represents an object in the model</td>
</tr>
<tr>
<td><a href="Sparqlpath/">Sparqlpath</a></td>
<td>A string encoding a SPARQL Property Path</td>
</tr>
<tr>
<td><a href="String/">String</a></td>
<td>A character string</td>
</tr>
<tr>
<td><a href="Time/">Time</a></td>
<td>A time object represents a (local) time of day, independent of any particular...</td>
</tr>
<tr>
<td><a href="Uri/">Uri</a></td>
<td>a complete URI</td>
</tr>
<tr>
<td><a href="Uriorcurie/">Uriorcurie</a></td>
<td>a URI or a CURIE</td>
</tr>
</tbody>
</table>
<h2 id="subsets">Subsets</h2>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>





                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
      
        
        <a href="about/" class="md-footer__link md-footer__link--next" aria-label="Next: About" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              About
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": ".", "features": ["content.tabs.link"], "search": "assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="assets/javascripts/bundle.5a2dcb6a.min.js"></script>
      
    
    
  </body>
</html>